frontend/ui/uimanager.lua

```diff
@@ -47,8 +48,42 @@ function UIManager:init()
         -- resume.
         self:_initAutoSuspend()
         self.event_handlers["Suspend"] = function(input_event)
-            self:_stopAutoSuspend()
-            Device:onPowerEvent(input_event)
+            if not self._power_off_triggered then
```

Comment:
Alright, that sounds like a good idea to me and I think I have a good understanding of how to implement that (and the same applies to the other remaining issues). I think I'll leave the `Resume` event as it is (sent when `device.screen_saver_mode` is true) and only send `PowerPress` and `PowerRelease` when `device.screen_saver_mode` is false. Then I'll move the current contents of the `Suspend` handler and `showPowerOffDialog` function from `UIManager` inside the new `PowerRelease` handler, conditionally executed depending on the duration between `PowerPress` and `PowerRelease`.

Is that what you were thinking? If it is, I should be able to revise the patch pretty soon.